TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/tests/test_source_cluster_multiversion.py

```diff
@@ -0,0 +1,139 @@
+    # Create a temporary services.yaml file based on the services.yaml spec
+    services_config = {
+        'source_cluster': {
+            'endpoint': base_url,
+            'allow_insecure': True,
+            'no_auth': {}
+        },
+        'target_cluster': {
+            'endpoint': "http://this_endpoint_does_not_exist:9200",
+            'allow_insecure': True,
+            'no_auth': {}
+        },
+        'snapshot': {
+            'snapshot_name': SNAPSHOT_NAME,
+            'fs': {
+                'repo_path': CLUSTER_SNAPSHOT_DIR
+            }
+        }
+    }
+
+    with tempfile.NamedTemporaryFile(mode='w', delete=False) as temp_config:
+        yaml.dump(services_config, temp_config)
+        temp_config_path = temp_config.name
+
+    logging.basicConfig(level=logging.INFO)
+
+    env = Environment(temp_config_path)
+    if version.flavor == "ELASTICSEARCH" and version.major_version == 5:
+        seed_data_with_types(env.source_cluster, DOC_COUNT)
+    else:
+        seed_data_without_types(env.source_cluster, DOC_COUNT)
+
+    setup_snapshot_repo_and_snapshot(env.source_cluster)
+
+    yield env
+    # Stop the container and clean up the temporary services.yaml file after tests complete
+    container.stop()
+    os.remove(temp_config_path)
+
+
+@pytest.mark.parametrize("env_with_source_container,refresh,json",
+                         itertools.product(SUPPORTED_SOURCE_CLUSTERS, [True, False], [True, False]),
+                         indirect=["env_with_source_container"])
+def test_cluster_cat_indices(env_with_source_container: Environment, refresh: bool, json: bool):
+    env = env_with_source_container
+    assert type(env.source_cluster) is Cluster
+    if not refresh:
+        # If no refresh, give the cluster a second to catch up with the document additions.
+        time.sleep(1)
+    result = clusters_.cat_indices(env.source_cluster, refresh=refresh, as_json=json)
+    if not json:
+        result = result.decode('utf-8')
+        result_lines = result.split('\n')
+        assert any(TEST_INDEX_NAME in line and str(DOC_COUNT) in line for line in result_lines)
+
+    else:
+        assert any(item['index'] == TEST_INDEX_NAME and int(item['docs.count']) == DOC_COUNT for item in result)
+
+
+@pytest.mark.parametrize("env_with_source_container", SUPPORTED_SOURCE_CLUSTERS, indirect=True)
+def test_connection_check(env_with_source_container: Environment):
+    env = env_with_source_container
+    assert type(env.source_cluster) is Cluster
+    result = clusters_.connection_check(env.source_cluster)
+    assert result.connection_established
+
+
+@pytest.mark.parametrize("env_with_source_container,deep_status_check",
+                         itertools.product(SUPPORTED_SOURCE_CLUSTERS, [False, True]),
+                         indirect=["env_with_source_container"])
+def test_snapshot_status(env_with_source_container: Environment, deep_status_check: bool):
```

Comment:
I've got beef with 'deep' vs 'normal' output, can change the behavoir to only have a singular output that we verify - or maybe only prioritize the deep scenario because will we fine a bug in the difference?